src/maptm/src/maptm_ovsdb.c

```diff
@@ -0,0 +1,262 @@
+ *
+ * Created: 05 February 2020
+ *
+ */
+#include "ovsdb_update.h"
+#include "ovsdb_sync.h"
+#include "ovsdb_table.h"
+#include "schema.h"
+#include <stdio.h>
+
+#include "maptm.h"
+
+#ifdef MAPTM_DEBUG
+#undef LOGI
+#define LOGI	printf
+#endif 
+
+/**
+ * Globals
+ */
+/* Log entries from this file will contain "OVSDB" */
+#define MODULE_ID LOG_MODULE_ID_OVSDB
+#define MAPTM_IFC_WAN "br-wan"
+
+ovsdb_table_t table_Netfilter;
+ovsdb_table_t table_Node_Config;
+struct ovsdb_table table_Node_State;
+static ovsdb_table_t table_Interface;
+static ovsdb_table_t  table_IP_Interface;
+ovsdb_table_t  table_Wifi_Inet_Config;
+ovsdb_table_t  table_DHCPv6_Client;
+ovsdb_table_t  table_DHCP_Client;
+ovsdb_table_t  table_IPv6_Address;
+/* To change to Enum  */
+int WanConfig =0; 
+
+bool maptm_update_mapt(bool enable)
+{
+	int rc=0;
+	json_t *where = NULL;
+	struct schema_Node_Config rec_config;
+	memset(&rec_config, 0, sizeof(rec_config));
+	rec_config._partial_update = true;
+
+	where = ovsdb_where_multi(
+		ovsdb_where_simple_typed(SCHEMA_COLUMN(Node_Config, module),"MAPTM", OCLM_STR),
+		ovsdb_where_simple_typed(SCHEMA_COLUMN(Node_Config, key),"maptParams", OCLM_STR),
+		NULL);
+	if(!where)
+	{
+		LOGE("check current mode Dual-Stack");
+		goto exit;
+	}
+	if (enable)
+	{
+		SCHEMA_SET_STR(rec_config.value, "{\"support\":\"true\",\"interface\":\"br-wan\"}");
+	}	
+	else
+	{
+		SCHEMA_SET_STR(rec_config.value, "{\"support\":\"false\",\"interface\":\"br-wan\"}");
+	}
+	rec_config.value_exists = true;
+
+	rc = ovsdb_table_update_where(&table_Node_Config, where, &rec_config);
+	if(rc != 1 )
+	{
+		LOGE("%s Could not update mapt table", __func__);
+		goto exit;
+	}
+
+	LOGD("%s Update mapt table", __func__);
+	return true;
+	exit:
+	return false;
+
+}
+
+
+bool maptm_persistent()
+{
+	bool ret = false ;
+	char mapt_support[10];
+	if (osp_ps_exists("MAPT_SUPPORT") )
+	{
+		if (osp_ps_get("MAPT_SUPPORT", mapt_support, sizeof("MAPT_SUPPORT")) != true)
+		{
+			LOGE("%s Cannot get MAPT_SUPPORT Value", __func__ );
+			return false;
+		}
+	}
+	else
+	{
+		snprintf(mapt_support,sizeof(mapt_support), "%s", "true");
+		if ((osp_ps_set("MAPT_SUPPORT", mapt_support)) != true)
+		{
+			LOGE("%s Cannot save mapt support through osp API",__func__);
+			return false;
+		}
+	}
+	LOGT("%s MAPT_Support= %s",__func__,mapt_support);
+
+	if(!strncmp(mapt_support,"true",sizeof("true")-1))
+	{
+		ret = maptm_update_mapt(true);
+	}
+	else
+	{
+		ret = maptm_update_mapt(false);
+	}
+
+	return ret;
+}
+
+
+bool maptm_get_supportValue(char * value)
+{
+	char str[64] ;
+	memset(str,0,sizeof(str));
+	strncpy(str,value,sizeof(str));
+	char delim[] = ":";
+
+	char *ptr = strtok(str, delim);
+
+	while (ptr != NULL)
+	{
+		if(!strncmp(ptr,"\"true\"",sizeof("\"true\"")) )
+		{
+			return true;
+	    }
+	    else if(!strncmp(ptr,"\"false\"",sizeof("\"false\"")) )
+	    {
+			return false;
+	    }
+		ptr = strtok(NULL, ",");
+	}
+
+	return false;
+}
```

Comment:
```suggestion
bool maptm_get_supportValue(char *value)
{
    char str[64];

    STRSCPY(str, value);

    char *ptr = strtok(str, ":");

    while (ptr != NULL)
    {
        if (!strcmp(ptr, "\"true\""))
        {
            return true;
        }
        else if (!strcmp(ptr, "\"false\""))
        {
            return false;
        }
        ptr = strtok(NULL, ",");
    }

    return false;
}

```

Reply:
Ok, but strcpy,strcmp .. generates a klocwork error, then in the sst quality process we must correct all  the klocwork issue